src/classes/RD2_ScheduleService.cls

```diff
@@ -0,0 +1,166 @@
+        rdsched.InstallmentFrequency__c = rd.InstallmentFrequency__c;
+        rdsched.InstallmentPeriod__c = rd.npe03__Installment_Period__c;
+        rdsched.PaymentMethod__c = rd.PaymentMethod__c;
+        rdsched.RecurringDonation__c = rd.Id;
+
+        if (!isFirstAndFifteenth(rd)) {
+            rdsched.DayOfMonth__c = rd.Day_of_Month__c;
+        }
+        else {
+            rdsched.DayOfMonth__c = FIRST;
+            schedulesToInsert.add(rdsched);
+            rdsched = rdsched.clone();
+            rdsched.DayOfMonth__c = FIFTEENTH;
+        }
+        schedulesToInsert.add(rdsched);
+
+        for (RecurringDonationSchedule__c rds : schedulesToInsert) {
+            rds.StartDate__c = calculateStartDate(rds, rd);
+        }
+
+        return this;
+    }
+
+    // TODO: Update schedule to reflect recurringDonation - W-036113 (Placeholder)
+    public RD2_ScheduleService updateSchedule(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdScheds) {
+        return this;
+    }
+
+    // TODO: Will return next Donation date - W-036251 (Placeholder)
+    public Date getNextDonationDate(RecurringDonationSchedule__c rdSched) {
+        return null;
+    }
+
+    // TODO: Will return an inner class (Placeholder - RDScheduleDates)
+    public List<RDScheduleDates> visualizeSchedule(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdScheds) {
+        return new List<RDScheduleDates>();
+    }
+
+    // TODO: Placeholder internal class used for visualization)
+    public class RDScheduleDates {
+        public npe03__Recurring_Donation__c recurringDonation;
+        public List<RecurringDonationSchedule__c> rdSchedules;
+        public Date donationDate;
+    }
+
+    /***
+    * @description Calculates start date for a recurring donation schedule
+    * @param rdSchedule Recurring Donation schedule
+    * @param rd Recurring Donation schedule
+    * @return Date
+    */
+    private Date calculateStartDate(RecurringDonationSchedule__c rdSchedule, npe03__Recurring_Donation__c rd) {
+        Date retDate;
+
+        Date minSD = rd.StartDate__c > rd.npe03__Date_Established__c ?
+                     rd.StartDate__c :
+                     rd.npe03__Date_Established__c;
+
+        retDate = minSD > Date.today() ?
+                  minSD :
+                  Date.today();
+
+        if (rdSchedule.InstallmentPeriod__c == MONTHLY || rdSchedule.InstallmentPeriod__c == FIRST_AND_FIFTEENTH) {
+            Integer lastDay = Date.daysInMonth(retDate.year(), retDate.month());
+            retDate = Date.newInstance(retDate.year(), retDate.month(), rdSchedule.DayOfMonth__c == LAST_DAY_OF_MONTH ?
+                                                                        lastDay :
+                                                                        Integer.valueOf(rdSchedule.DayOfMonth__c));
+            retDate = retDate >= Date.today() ?
+                      retDate :
+                      retDate.addMonths(1);
```

Comment:
As noted in the previous comment (that seems to be hidden now), I'm questioning whether we should recalculate the start date at all. The A/C's on the work item were updated with questions. We'll have an exact answer once Mary and I discuss.